AwsCryptographicMaterialProviders/runtimes/java/build.gradle.kts

```diff
@@ -88,6 +153,41 @@ publishing {
         artifact(tasks["javadocJar"])
         artifact(tasks["sourcesJar"])
 
+        // Since we ship the MPL bundled with our generated dependencies they should not be included in the generated pom.xml
+        // however; we also use additional dependencies runtime dependencies that are needed in order to properly run the mpl.
+        // When you bundle a shadow jar you don't need to include any dependencies in the pom.xml since everything is on the jar, but since
+        // we are selective so we have to "manually" write our own pom file.
+        pom.withXml {
+            var dependencyManagementNode = asNode().appendNode("dependencyManagement").appendNode("dependencies").appendNode("dependency")
+            var sdkPomDependency = configurations.implementation.get().dependencies.find { dependency ->  dependency.group.equals("software.amazon.awssdk") && dependency.name.equals("bom") }
+            if (sdkPomDependency != null) {
+                dependencyManagementNode.appendNode("groupId", sdkPomDependency.group)
+                dependencyManagementNode.appendNode("artifactId", sdkPomDependency.name)
+                dependencyManagementNode.appendNode("version", sdkPomDependency.version)
+                dependencyManagementNode.appendNode("type", "pom")
+                dependencyManagementNode.appendNode("scope", "import")
+            }
+
+            var dependenciesNode = asNode().appendNode("dependencies")
+            configurations.implementation.get().dependencies.forEach {
+                // we don't want to include generated dependencies
+                if (!(it.name.equals("StandardLibrary") ||
+                                it.name.equals("AwsCryptographyPrimitives") ||
+                                it.name.equals("ComAmazonawsKms") ||
+                                it.name.equals("ComAmazonawsDynamodb") ||
+                                it.name.equals("bom"))
+                ) {
+                    var dependencyNode = dependenciesNode.appendNode("dependency")
+                    dependencyNode.appendNode("groupId", it.group)
+                    dependencyNode.appendNode("artifactId", it.name)
+                    if (!it.version.isNullOrEmpty()) {
+                        dependencyNode.appendNode("version", it.version)
+                    }
+                    dependencyNode.appendNode("scope", "runtime")
+                }
+            }
+        }
```

Comment:
Issue:
I know I am too late, and this has already been merged.
But Kotlin Scripts support methods.

This repeat logic SHOULD have been moved into a common method.

Reply:
addressed in #44